src/lib.rs

```diff
@@ -1732,45 +1808,48 @@ impl GooseAttack {
-                    if goose_attack_run_state.shutdown_after_stop {
-                        self.set_attack_phase(&mut goose_attack_run_state, AttackPhase::Shutdown);
-                    } else {
-                        // Print metrics, if enabled.
-                        if !self.configuration.no_metrics {
-                            println!("{}", self.metrics);
+
+                    if goose_attack_run_state.active_users == 0 {
+                        // Stop any running GooseUser threads.
+                        self.stop_attack().await?;
+                        // Collect all metrics sent by GooseUser threads.
+                        self.sync_metrics(&mut goose_attack_run_state, true).await?;
+                        // Record last users for users per second graph in HTML report.
+                        if let Some(started) = self.started {
+                            self.graph_data.record_users_per_second(
+                                goose_attack_run_state.active_users,
+                                started.elapsed().as_secs() as usize,
+                            );
+                        };
+                        // The load test is fully stopped at this point.
+                        self.metrics
+                            .history
+                            .push(TestPlanHistory::step(TestPlanStepAction::Finished, 0));
+                        // Write an html report, if enabled.
+                        self.write_html_report(&mut goose_attack_run_state).await?;
+                        // Shutdown Goose or go into an idle waiting state.
+                        if goose_attack_run_state.shutdown_after_stop {
+                            self.set_attack_phase(
+                                &mut goose_attack_run_state,
+                                AttackPhase::Shutdown,
+                            );
+                        } else {
+                            // Print metrics, if enabled.
+                            if !self.configuration.no_metrics {
+                                println!("{}", self.metrics);
+                            }
+                            self.set_attack_phase(&mut goose_attack_run_state, AttackPhase::Idle);
                         }
-                        self.set_attack_phase(&mut goose_attack_run_state, AttackPhase::Idle);
                     }
                 }
                 // By reaching the Shutdown phase, break out of the GooseAttack loop.
-                AttackPhase::Shutdown => {
-                    self.update_duration();
-                    break;
-                }
+                AttackPhase::Shutdown => break,
             }
 
             // Record current users for users per second graph in HTML report.
             if let Some(started) = self.started {
                 self.graph_data.record_users_per_second(
-                    self.metrics.users,
+                    goose_attack_run_state.active_users,
```

Comment:
Other graphs look OK, but this one seems odd. Is this variable always storing the number of users at the time or the target number for the step?

Reply:
In my case it happened after the first `Increase` step when the `Maintain` step started. After that is seems OK. 

The load test was started with `--test-plan "5,10;5,20;20,10;20,30;100,10;20,10;20,30;0,20"`